### DISCUSSION TITLE
Oh-my-posh configured correctly in windows terminal, doesn't work in vscode

### DISCUSSION THREAD

**Discussion by andreapesci** [JanDeDobbeleer/oh-my-posh#1011] [Troubleshoot]
Hello! 
The title summarize my problem.
I have configured oh-my-posh for windows terminal and it works just fine, altough on vscode i get this problem: 

`oh-my-posh: The term 'oh-my-posh' is not recognized as a name of a cmdlet ...`

How can i solve it?

**Comment by JanDeDobbeleer** [comment]
How did you install oh-my-posh?

**Reply by andreapesci** [reply]
I installed it using winget.

`winget install JanDeDobbeleer.OhMyPosh`

then put this into $PROFILE:

`oh-my-posh --init --shell pwsh --config ~/jandedobbeleer.omp.json | Invoke-Expression`

**Reply by JanDeDobbeleer** [reply]
Did you restart vscode or your machine?

**Reply by andreapesci** [reply]
i restarted vscode multiple times, since i didnt think a whole pc restart was needed... but apparently that fixed the problem! sorry for wasting your time -_-'

**Reply by JanDeDobbeleer** [reply]
Windows and PATH updates are a pain 😬

**Reply by ghost** [reply]
Restarting windows also fixed the issue for me 😄

**Reply by eersonmez** [reply]
Restarting explorer.exe was sufficient (vs a full restart)
1. Close all VS Code instances
2. Task Manager > Details > find and shutdown explorer.exe
3. Task Manager > File > Run new task > explorer.exe.

**Comment by yeonghunki** [comment]
In my case it was because vscode on windows 11 couldn't load user environment variables after the version update.

simply closing all the vscode windows and reopening it fixed the issue.

**Comment by AshleyFang** [comment]
issue：oh-my-posh work well in windows terminal √ vscode ×

try：setting PATH × restart vscode ×

success：
1 setting PATH
2 restart  computer√

**Comment by Fred-Vatin** [comment]
For me, the issue was because of a previous VSCode settings that used to work but not in my new windows install or the last VSCode.
I just deleted this settings because if `${workspaceFolder}`, the `PATH` is not substituted. Only use this in a workspace settings where you know all environment exist.
```json
"terminal.integrated.env.windows": {
  "PATH": "${env:PATH};${workspaceFolder}\\node_modules\\.bin;${env:PNPM_HOME}"
},
```

But this setting should be set:
```json
"terminal.integrated.defaultProfile.windows": "PowerShell",
```

I opened an issue [here](https://github.com/microsoft/vscode/issues/233495#issue-2646380894). I encourage you to vote for it.
